includes/rest-api/controllers/class-nps-controller.php

```diff
@@ -116,8 +117,10 @@ public function upsert_nps_response( \WP_REST_Request $request ) {
 		$data      = $request->get_json_params();
 		$survey_id = $request->get_param( 'survey_id' );
 
+		$verifies = wp_verify_nonce( $data['nonce'], Crowdsignal_Forms_Nps_Block::NONCE );
```

Comment:
Also here, we should probably have a custom `verify_nonce` method which basically does `$nonce === $this->create_nonce()`. Just to make sure if any filters were applied they also apply for the verification.